lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java

```diff
@@ -68,11 +68,14 @@ public FileEntry next()
         }
 
         String root = listingPath.toUri().getPath();
+        if (!root.endsWith("/")) {
+            root += "/";
```

Comment:
`listingPath` is const, so is `root`.
Can this logic be applied once in the ctor?

Also, add a comment why append slash, what cases this is needed for.

Reply:
Not sure but it does not look so.

The issue I observed was coming from the fact that here:
https://github.com/trinodb/trino/blob/5b8e06a9ecfaf5b2e7255419b787c00f9c0c345d/plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/DirectoryListingFilter.java#L71

we got parent directory of a file without trailing slash. While `prefix` passed to `DirectoryListingFilter` was with slash - and filtering did not work as expected.

Will this code path be affected?